public/Stations.php

```diff
@@ -82,4 +82,16 @@ public function getRelatedOnramps($id) {
   public function getDetectors($id) {
     return Detector::fetchForStation($id);
   }
+
+  /** Will take a coordinate and return the closest stationid
+   *
+   * Will take a coordinate and return the closest stationid
+   *
+   * @param $coord
+   * @return null
+   */
+  public function getStationfromCoord($coord){
```

Comment:
Best practice: Name should be getStationFromCoord. (Capital F in From)

Also, this looks a lot like model logic... Since it doesn't have an @url that maps it to a REST endpoint, it probably doesn't need to be in the controller and should instead be in the entity.